davenjeru/authors-haven-playground
# Maintainability config

#### What does this PR do?
 - Adds the maintainability configurations

#### Description of Task to be completed?
- Add a [pull request template file](https://github.com/davenjeru/authors-haven-playground/blob/maintainability-config/PULL_REQUEST_TEMPLATE.md) to make pull request description writing easier.
- Add [eslint](https://github.com/davenjeru/authors-haven-playground/blob/maintainability-config/.eslintrc.json) to check the code style conventions which should be similar to those of Airbnb.
- Add [.eslintignore](https://github.com/davenjeru/authors-haven-playground/blob/maintainability-config/.eslintignore) to ignore the [serviceWorker.js](https://github.com/davenjeru/authors-haven-playground/blob/maintainability-config/src/serviceWorker.js) file which was generated by create-react-app command.
- Configure [Hound](https://houndci.org) to check the project for code style violations.
- Configure [TravisCI](https://travis-ci.org/) to run our tests.
- Configure [CodeClimate](https://codeclimate.com/) to check the project for code style violations
- Fix eslint issues in [App.js](https://github.com/davenjeru/authors-haven-playground/blob/maintainability-config/src/App.js)

#### How should this be manually tested?
- Visit this [link]() to view the CodeClimate score.
- Visit this [link]() to view the TravisCI build.
- Check the comments from Hound on this Pull Request.
